### DISCUSSION TITLE
Add Cache Response using Web Cache API

### DISCUSSION THREAD

**Discussion by kresnasatya** [oven-sh/bun#15845] [Ideas]
I see a neighbour of Bun, Deno has implement [Cache Response](https://docs.deno.com/api/web/~/Cache) using [Web Cache API](https://deno.com/blog/deploy-cache-api#using-the-cache-api). I think if Bun can catch up this it will be great.

## Background

I create web app that utilize Cache Response using SvelteKit & Cloudflare Pages (functions). The reason I choose Cloudflare because [SvelteKit and Cloudflare Pages communicate using Web API standards](https://developers.cloudflare.com/pages/framework-guides/deploy-a-svelte-site/#functions-setup), it makes more easier to implement a Cache Response.

Btw, I'm almost keep it in two repositories and replace the Worker (plain) with Hono because it supports [Cache Middleware](https://hono.dev/docs/middleware/builtin/cache). But, I think it makes more complicated if I do this way.

Also, the Cache Middleware of Hono framework currently support in Cloudflare and Deno.

## Syntax

The GET function comes from SvelteKit but the `caches` comes from Cloudflare.

```js
import { allTags } from '$lib/github';

export async function GET({ request, url }) {
    const isCloudflare = typeof caches !== 'undefined';
    
    if (isCloudflare) {
        const cacheKey = new Request(url.toString(), request);
        const cache = caches.default;
        let response = await cache.match(cacheKey);
        if (response) return response;
    }

    let json = await allTags('laravel/laravel');

    let response = new Response(JSON.stringify(json));

    response.headers.set('Cache-Control', 'public, max-age=3600');
    response.headers.set('Content-Type', 'application/json');
    response.headers.set('Access-Control-Allow-Origin', '*');
    response.headers.set('Access-Control-Allow-Methods', 'GET');

    // Put the response in the cache
    if (isCloudflare) {
        const cache = caches.default;
        const cacheKey = new Request(url.toString(), request);
        cache.put(cacheKey, response.clone());
    }

    return response;
}
```

**Comment by sntran** [comment]
I'd second this. As another note, `undici` also [has cache support](https://github.com/nodejs/undici/tree/main/lib/web/cache).

Interestingly, Deno's support of Cache API uses SQLite, which bun already supports natively.
